src/main/java/angel/Parser.java

```diff
@@ -0,0 +1,265 @@
+ */
+public class Parser {
+    private static final DateTimeFormatter INPUT_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    /**
+     * Parses the input command and its parameters.
+     *
+     * @param input The full user input.
+     * @return A Command object containing the command type and its parameters.
+     * @throws InvalidCommandException If the command is not valid.
+     */
+    public static Command parseCommand(String input) throws InvalidCommandException {
+        String[] parts = input.split(" ", 2);
+        String command = parts[0];
+
+        switch (command) {
+        case "bye":
+            return new Command(command, null);
+        case "list":
+            return new Command(command, null);
+        case "find":
+            return parseFindCommand(parts);
+        case "todo":
+            return parseTodoCommand(parts);
+        case "deadline":
+            return parseDeadlineCommand(parts);
+        case "event":
+            return parseEventCommand(parts);
+        case "mark":
+        case "unmark":
+        case "delete":
+            return parseIndexCommand(parts, command);
+        default:
+            throw new InvalidCommandException("Unknown command: " + command + ". Please enter a valid command (e.g., list, todo, deadline, event, mark, unmark, delete, bye, find).");
+        }
+    }
+
+    /**
+     * Parses the command for searching tasks by a keyword.
+     *
+     * @param parts The parts of the user input split by space.
+     * @return A Command object for the find task.
+     * @throws InvalidCommandException If the find keyword is missing.
+     */
+    private static Command parseFindCommand(String[] parts) throws InvalidCommandException {
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
+            throw new InvalidCommandException("The find command requires a keyword. Please specify a keyword after 'find'. Example: 'find book'.");
+        }
+        return new Command("find", parts[1].trim());
+    }
+
+    /**
+     * Parses the command for creating a todo task.
+     *
+     * @param parts The parts of the user input split by space.
+     * @return A Command object for the todo task.
+     * @throws InvalidCommandException If the todo description is empty.
+     */
+    private static Command parseTodoCommand(String[] parts) throws InvalidCommandException {
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
+            throw new InvalidCommandException("The description of a todo cannot be empty. Please specify a task after 'todo'.");
+        }
+        return new Command("todo", parts[1].trim());
+    }
+
+    /**
+     * Parses the command for creating a deadline task.
+     *
+     * @param parts The parts of the user input split by space.
+     * @return A Command object for the deadline task.
+     * @throws InvalidCommandException If the deadline description or date/time is missing or incorrectly formatted.
+     */
```

Comment:
I like how you have framed most of your javaDoc descriptions. Simple and easy to read